clarsach/spectrum.py

```diff
@@ -12,8 +12,56 @@
 
 __all__ = ['XSpectrum']
 
+def calculate_flux_spectrum(spec, counts=None):
+    """
+    Convert a spectrum from units of photon counts to
+    units of photon flux.
+
+    **Warning**: You should do this for plotting *only*.
+    Converting from flux to counts, as is done in the detector,
+    involves a convolution with the response matrix of the detector.
+    Deconvolving this correctly is very hard, and not done in this
+    function. This is for visualization purposes only!
+
+    Parameters
+    ----------
+    spec : clarsach.XSpectrum object
+        An XSpectrum object containing the X-ray spectrum
+
+    counts : iterable
+        An array with counts to be converted to the flux
+        If this is None, use the `counts` attribute of the `spec`
+        object.
+
+    Returns
+    -------
+    flux_spectrum:
+        The flux spectrum
+
+    """
+
+    # make a flat spectrum so that I can integrate
+    # ARF and RMF only
+    flat_model = np.ones_like(spec.counts) * (spec.rmf.energ_hi -
+                                              spec.rmf.energ_lo)
+
+    # now apply ARF to the flat model
+    m_arf = spec.arf.apply_arf(flat_model)
```

Comment:
Use `spec.apply_resp(flat_model)` here instead of `apply_arf` and `apply_resp`.

`apply_resp` takes into account fracexpo.  It is not specific to Chandra.  For ARF files where the "FRACEXPO" column does not exist, fracexpo is set to 1.0 and doesn't really do anything.